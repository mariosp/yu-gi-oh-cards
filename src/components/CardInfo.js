import React from 'react';


export const CardInfo = (props) =>
    (
        <div className="cardinfo">
            
        <img className="img" src={"http://52.57.88.137/api/card_image/"+props.param.name} />

        <div className="infobox">
        <div className="desc">
        <h3>Description</h3>
        <div className="text">
        {props.param.text}
        </div>
        </div>

        <div className="details">
        <h3>Details</h3>
        <div className="text">
        {props.param.card_type ? <li>Card Type: {props.param.card_type}</li>:''}
        {props.param.family ? <li>Family: {props.param.family}</li>:''}
        {props.param.type ? <li>Type: {props.param.type}</li>:''}
        {props.param.property ? <li>Property: {props.param.property}</li>:''}
        {props.param.property ? <li>Property: {props.param.property}</li>:''}
        {props.param.level ? <li>Level: {props.param.level}</li>:''}
        {props.param.atk ? <li>Attack: {props.param.atk}</li>:''}
        {props.param.def ? <li>Defence: {props.param.def}</li>:''}
        </div>
        
        </div>

        </div>

            
        </div>
    );




export default CardInfo;