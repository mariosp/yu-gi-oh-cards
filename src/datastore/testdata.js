

export default [{
    atk: null,
    card_type: "spell",
    def: null,
    family: null,
    level: null,
    name: "Monster Gate",
    property: "Normal",
    text: "Tribute 1 monster; reveal cards from the top of your Deck until you reveal a monster that can be Normal Summoned. Special Summon it, also send the other revealed cards to the Graveyard.",
    type: null
    
}]