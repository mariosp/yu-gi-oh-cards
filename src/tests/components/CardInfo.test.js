import React from 'react'
import {shallow} from 'enzyme';
import CardInfo from '../../components/CardInfo';
import testdata from '../../datastore/testdata'

test('should render CardInfo (main content) with all card details', ()=>{

    const wrapper = shallow(<CardInfo param={testdata[0]} />);
    expect(wrapper).toMatchSnapshot();
    
    
    });