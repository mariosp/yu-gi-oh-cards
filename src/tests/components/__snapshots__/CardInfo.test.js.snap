// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CardInfo (main content) with all card details 1`] = `
<div
  className="cardinfo"
>
  <img
    className="img"
    src="http://52.57.88.137/api/card_image/Monster Gate"
  />
  <div
    className="infobox"
  >
    <div
      className="desc"
    >
      <h3>
        Description
      </h3>
      <div
        className="text"
      >
        Tribute 1 monster; reveal cards from the top of your Deck until you reveal a monster that can be Normal Summoned. Special Summon it, also send the other revealed cards to the Graveyard.
      </div>
    </div>
    <div
      className="details"
    >
      <h3>
        Details
      </h3>
      <div
        className="text"
      >
        <li>
          <span>
            Card Type:
          </span>
           
          spell
        </li>
        <li>
          <span>
            Property:
          </span>
           
          Normal
        </li>
      </div>
    </div>
  </div>
</div>
`;
